[quick-start]simple-solution-for-node-setup

pieceofr opened the pull request:
The documentation is a quick-start of bitmark blockchain using bitmark-node docker container. The way of writing documentation is to minimize steps. The reference is Libra's documentation(https://developers.libra.org/) The documentation style is according to the discussion of bitmark-node docker container feebback.   
The TAs are the engineers who are capable of setting up docker and network. The approach to improve document is to get approval by an engineer (Linzy) then later other engineers.

nodestory committed:
explain bitmarkd storage

Commit:
[quick-start]simple-solution-for-node-setup

Commit:
[quick-start]simple-solution-for-node-setup

nodestory commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
```diff
@@ -1 +1,67 @@
++ input your BTC and LTC address for rewarding in a pop-up screen
+![Input your BTC&LTC Address](https://i.imgur.com/IRTlyjY.jpg?1)
+
+### Step 2: Run bitmarkd node and recorderd
```
```suggestion
### Step 2: Run bitmarkd and recorderd
```
bitmarkd node = `bitmarkd` + `recorderd`?

vienlt replied:
Can we explain what the script does?

nodestory commented:
@pieceofr Would you mind doing a spell check?

nodestory commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
```diff
@@ -1 +1,67 @@
 #  [Learning Bitmark / Quick start] - Simple solution for node setup
+
+### Start your bitmark-node
+This section will quide you to start a bitmark docker node. The **docker** node, also called **bitmark-node**, is a bitmark full-node. A full-node can generate blocks, mine the blocks and do verification of payments. 
```
Does `bitmark-node` only refer to the node running inside a docker container?

vienlt replied:
Do you think we should introduce a bit about the bitmark node in general first, then back to the content of the section? 
Just for reference, here is the text from the technical writer for the section `Run a node` which will be removed from the website:

>Unlike conventional property systems that rely on centralized gatekeepers, the Bitmark blockchain is an open and transparent property system that is strengthened through the active participation of anyone on the Internet. You can support the integrity of Bitmark’s open-source blockchain by running the Bitmark node software and becoming a member of its peer-to-peer network. You will earn money and property rewards by doing so.

After that introduction, we could mention this section, such as:
>There are different ways to run a Bitmark node, in this section, we introduce a simple way to do so with running a node contributed as a docker container image. ...

pieceofr replied:
Is this ok , anything to add !? 
Start your bitmark node
A user can run a bitmark node in several ways. This section will guide you to start a bitmark node using docker container. Inside a bitmark node docker container, there are bitmarkd, recorderd, payment system and a user interface to control and monitor services.

nodestory commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
```diff
@@ -1 +1,67 @@
+    + After bitmarkd is running, run a recorderd by click "start button" of "Recorder Node (recorderd)"
+![startup screen](https://i.imgur.com/aeONALb.jpg)
+
++ Running screen
```
Shall we teach them how to identify if the node is successfully or not?

vienlt replied:
How can users access the screen?

pieceofr replied:
@nodestory As our discussion, I will add a check section in the bottom. 
@vienlt ,  in step 1
Step 1: Open A Node-Management Webpage
view our node-mangement webpage on http://127.0.0.1:9980 in a browser

nodestory commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
```diff
@@ -1 +1,67 @@
+
+### Step 0: Preparation
++ Have your **Public IP** ready
++ Setup port-forwarding if nessasary
```
Readers might not remember what makes it necessary to set up port forwarding?

vienlt replied:
1. I think the list under `Assumptions` is more about `Prerequites`. 
2. Why do we need `git` installed?

pieceofr replied:
@nodestory as discussion change to
```+ You may need to setup your network, ie. port forwarding, in order to allow other bitmark nodes to communicate with your bitmark node.```
@vienlt  
1. I can change to "Prerequisites" . Libra uses Assumptions. They may have a subtle difference but I think we don't need to argue and maybe leave it to Technical writer. 
2. Remove git

pieceofr replied:
@vienlt  You also may consider We have a preparation step so is a prerequisite a better word?  I'll let you decide

vienlt reviewed (commented):
In general, I think we should add a short intro text for each heading. And some following concepts should be explained/ mentioned briefly:

- What is a bitmark node?
- This section is about running a bitmark node using the docker container image.
- What does bitmarkd do?
- What does recorderd do? Why running recorderd is an option.
- When the user successfully run a node.

pieceofr committed:
Update learning-bitmark/quick-start/simple-solution-for-node-setup.md

bitmarkd node is bitmarkd. This is first time I mention bitmarkd so I think we should add a node. I will remove it

Co-Authored-By: nodestory <linzy.hu@gmail.com>

pieceofr committed:
Update learning-bitmark/quick-start/simple-solution-for-node-setup.md

bitmarkd node is bitmarkd. This is first time I mention bitmarkd so I think we should add a node. I will remove it

Co-Authored-By: nodestory <linzy.hu@gmail.com>

Commit:
- Add imgs.

araujobsd committed:
Merge pull request #36 from bitmark-inc/contributing-to-bitmark

- Add imgs for contributing-to-bitmark.

pieceofr committed:
spelling checking and more descriptions

pieceofr committed:
spelling checking and more descriptions

pieceofr committed:
[ Learning Bitmark / Quick start ] Add network check information

pieceofr committed:
[ Learning Bitmark / Quick start ] Add network check information

pieceofr committed:
[ Learning Bitmark / Quick start ] modify Network Check Part

pieceofr committed:
[ Learning Bitmark / Quick start ] modify Network Check Part

vienlt commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
```diff
@@ -1 +1,96 @@
++ you [log in](https://docs.docker.com/engine/reference/commandline/login/) your docker with [docker hub]((https://hub.docker.com/)) account
++ you have and know your **public IP** address
++ you have **a bitcoin and a litecoin address** for receiving mining reward
++ you may need to setup your network to allow other bitmark nodes to communicate with your bitmark node. You can check more information on "Important Notice On Network" below.
```
Please capitalize the first character of a bullet except for the special cases. In this section, I think the `you` should be `You`.

vienlt replied:
I couldn't find the `"Important Notice On Network"`, could you link the text to the corresponding section?

nodestory replied:
- your environment is open for incoming connections through port 2136

And I see we try to guide readers to `Important Network Notes` more than once, and I guess readers are very likely to be forced to scroll down to the bottom for that section. Does it make sense that we won't guide the readers to `Important Network Notes` section until they find the node is not running properly?

And are there other common problems when running a node? Could `Important Network Notes` be a more general troubleshooting section?

pieceofr replied:
changed as request

vienlt commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
```diff
@@ -1 +1,96 @@
+    + port 	**2130** for Bitmark node RPC server
+
+You can check more information on "Important Notice On Network" below.
+
```
`port` to `Port` for each bullet.

vienlt replied:
`from internet` to `from the Internet`.

vienlt replied:
>I couldn't find the "Important Notice On Network", could you link the text to the corresponding section? And do you mean Important Network Notes?

pieceofr replied:
have add link

vienlt commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
````diff
@@ -1 +1,96 @@
+### Step 1: Download and Run Script
+
+Downloaded script will help you to manage bitmark node container. The  installation of the script will pull down a latest bitmark node container image from docker hub and then start a bitmark node container for you.  To check what functions the script supports and how to run it, simply run ``` bash install-node-linux-mac.sh```.  or ```bitmarkNode-HyperV.bat```.
+
````
Suggestions:

- `Downloaded script` to `The script`

- `will pull down a latest...` to `will pull down the latest...`

Personally I think the last sentence is a bit confusing because the `bash` command is to run the script instead of checking how to run it.

vienlt commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
````diff
@@ -1 +1,96 @@
+    + [download scripts for linux and mac](https://bitmark-node-docker-scripts.s3-ap-northeast-1.amazonaws.com/install-node-linux-mac.sh)
+    + execute bash install-node-linux-mac.sh [Your Public IP]. When run the script without providing [Your Public IP], the script will find your IP from Internet
+
+        ```bash install-node-linux-mac.sh 123.123.123.123```
````
A suggestion about the format:

- Run the downloaded script:
   ```shell
      $ bash install-node-linux-mac.sh <Your Public IP>
   ````
    > **Note:** In case the Public IP isn't provided, the script finds the IP from the Internet.

pieceofr replied:
@vienlt  do you think it is a good idea to tell the user we will find the IP for them and that IP is not workable in many cases ?

pieceofr replied:
@shannona  Can you give us some opinion?

vienlt replied:
@pieceofr : I just change the format of the instructions without changing any content. What is the concern here?

pieceofr replied:
@vienlt  Got it. Now No concern

vienlt commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
````diff
@@ -1 +1,96 @@
+
+        ```bash install-node-linux-mac.sh 123.123.123.123```
+    + follow script instruction and select "1)Installation"
+
````
`follow script instruction` to `Follow the script instructions`

pieceofr replied:
ok

vienlt commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
````diff
@@ -1 +1,96 @@
+    + execute bitmarkNode-HyperV.bat [Your Public IP]. When run the script without providing [Your Public IP], the script find your IP from Internet
+
+        ``` bitmarkNode-HyperV.bat 117.166.111.123```
+    + follow script instruction and select "1)Installation"
````
Same comments as in `Linux and Mac users` section.

pieceofr replied:
ok

vienlt commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
```diff
@@ -1 +1,96 @@
+## Management Panel
+
+### Step 1: Open A Node-Management Webpage 
++ view our node-mangement webpage on http://127.0.0.1:9980 in a browser
```
Suggestion:
From:  `view our node-mangement webpage on http://127.0.0.1:9980 in a browser` 
To: `Access the node management webpage at http://127.0.0.1:9980`

pieceofr replied:
ok

nodestory commented:
@vienlt I'm wondering how we should talk to readers.

option 1 `A user can run a bitmark node in several ways.`
option 2 `You can run a bitmark node in several ways.`

vienlt commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
```diff
@@ -1 +1,96 @@
+![startup screen](https://i.imgur.com/aeONALb.jpg)
+
++ Running screen
+    + wait until 5 connections to establish a distribute network
```
Do you mean `Wait until 5 nodes connected to establish a distributed network`?

pieceofr replied:
will change to Linzy's suggestion

vienlt commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
```diff
@@ -1 +1,96 @@
+
++ Running screen
+    + wait until 5 connections to establish a distribute network
+    + "Current Block" shows progress of receiving blocks
```
What do you think about the following text?
`The "Current Block" shows the progress of blocks synchronization of the node.`

pieceofr replied:
good! will change to it

vienlt reviewed (commented):
Can we capitalize all the first characters of a sentence or a bullet point?

nodestory commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
```diff
@@ -1 +1,96 @@
++ Make sure following ports are accessible from internet
+    + port	**2136** for connecting to other peer bitmarkd nodes
+    + port	**2135** for publishing blockchain events
+    + port 	**2130** for Bitmark node RPC server
```
Maybe we don't require ports 2130 and 2135 to be publicly accessible. Users can use localhost to subscribe events and send rpc commands with their own node only.

pieceofr replied:
ok

nodestory commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
```diff
@@ -1 +1,96 @@
+![startup screen](https://i.imgur.com/aeONALb.jpg)
+
++ Running screen
+    + wait until 5 connections to establish a distribute network
```
```suggestion
    + wait for 5 connections to join the network
```

pieceofr replied:
ok

vienlt commented:
> @vienlt I'm wondering how we should talk to readers.
> 
> option 1 `A user can run a bitmark node in several ways.`
> option 2 `You can run a bitmark node in several ways.`

@nodestory: I personally would like to avoid both options. My option is 
> There are several ways to run a bitmark node.

pieceofr committed:
minimized requirement and cap first letter of an item

pieceofr committed:
minimized requirement and cap first letter of an item

pieceofr commented:
@vienlt I'm wondering how we should talk to readers.




option 1 A user can run a bitmark node in several ways.
option 2 You can run a bitmark node in several ways.
@nodestory: I personally would like to avoid both options. My option is
There are several ways to run a bitmark node.

@vienlt @nodestory  Why not leave it to technical writer?  XD

pieceofr committed:
modified according to PR

pieceofr committed:
modified according to PR

pieceofr commented:
@shannona  We have different ways to use words. This is the first line in the first paragraph. Can you give us some opinion?  

@vienlt I'm wondering how we should talk to readers.
option 1 A user can run a bitmark node in several ways.
option 2 You can run a bitmark node in several ways.
@nodestory: I personally would like to avoid both options. My option is
(option 3) There are several ways to run a bitmark node.

vienlt commented:
> @vienlt I'm wondering how we should talk to readers.
> 
> option 1 A user can run a bitmark node in several ways.
> option 2 You can run a bitmark node in several ways.
> @nodestory: I personally would like to avoid both options. My option is
> There are several ways to run a bitmark node.
> 
> @vienlt @nodestory Why not leave it to technical writer? XD

It is ok to let Shannon edit, however, why don't we edit it by ourselves when we can?

Commit:
- Add circleci with igor, from now on we gonna try to catch as
  much errors as possible in our documentation.

Commit:
- Add recursivety using a simple shell script to parse all files
  from a directory tree.

Commit:
- we don't need to call "sh" for that script

Commit:
- We don't need *

Commit:
- Fix typos and contractions found by igor(1).

pieceofr committed:
[Quick start] -Simple solution for node setup - change IP format

pieceofr committed:
[Quick start] -Simple solution for node setup - change IP format

Commit:
appendix-bitmark-vs-eth

Commit:
fix-igor-circle

vienlt committed:
Merge pull request #41 from bitmark-inc/appendix-eth-comparison

Appendix Bitmark vs. ETH comparison

shannona commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
```diff
@@ -1 +1,85 @@
+## Start your bitmark node
+
+A user can run a bitmark node in several ways. This section will guide you to start a bitmark node using docker container. Inside a bitmark node docker container, there are bitmarkd, recorderd, payment system and a user interface to control and monitor services.
+
```
If you're going to mention multiple ways, it's probably worth mentioning what the other methods are. It's definitely worth noting this is the simplest, quickest, and most reliable setp method.

shannona replied:
And I see I missed your question here, about which is best: "option 1 A user can run a bitmark node in several ways.
option 2 You can run a bitmark node in several ways.
@nodestory: I personally would like to avoid both options. My option is
There are several ways to run a bitmark node."

The last one is passive voice, where you don't have an active subject. I would usually avoid it unless there are no options (and sometimes there aren't).

The other two are both fine. Saying "A user" is more formal and "you" is more casual. I personally like "you" because it's very direct. The biggest thing is just to be consistent. If you regularly address the readers as "a user" (or other noun), don't use "you", and if you regularly use "you", use it whenever you're directly addressing the readers. So, make a style decision, and we can just ensure that it's used.

shannona commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
```diff
@@ -1 +1,85 @@
+## Prerequisites
+
++ [Docker](https://docs.docker.com/install/) is installed in your system
+    +  Windows user run docker on Hyper-V
```
This seems confusing: readers won't necessarily understand it, and it's already covered in the Docker docs (in fact, it looks like the Docker install automatically turns on Hyper-V)

pieceofr replied:
@shannona  some widows do not support hyper-v ie, home-edition but I can not figure out them all. It depends on users to find out themselves. What is your suggestion for this situation?

shannona commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
````diff
@@ -1 +1,85 @@
+### Step 1: Download and Run Script
+
+Downloaded script will help you to manage bitmark node container. The  installation of the script will pull down a latest bitmark node container image from docker hub and then start a bitmark node container for you.  To check what functions the script supports and how to run it, simply run ``` bash install-node-linux-mac.sh```.  or ```bitmarkNode-HyperV.bat```.
+
````
You should say which system each of these commands runs on:
(on Mac and Linux)
(on Windows)

pieceofr replied:
i see

shannona commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
```diff
@@ -1 +1,85 @@
+
++  Linux and mac users
+    + [Download scripts for linux and mac](https://bitmark-node-docker-scripts.s3-ap-northeast-1.amazonaws.com/install-node-linux-mac.sh)
+    + Execute bash install-node-linux-mac.sh [Your Public IP]. When run the script without providing [Your Public IP], the script will find your IP from Internet
```
In-line commands should usually be marked for clarity: `bash install-node-linux-mac.sh [Your Public IP]`

shannona replied:
If you're just going to list it in a full example (as you do here), then you can probably have something shorter in the description, e.g.:

* Execute the install script using `bash`, including your Public IP address. If you do not know your IP address, you can omit that, and the script will try to determine it from the internet:

        ```bash install-node-linux-mac.sh  <Your Public IP>```

pieceofr replied:
Can you give me an example of shortening the description?  It is very short for my point of view. What do you mean in-line command should be marked for clarity?   I have a ``` ``` (Comment on that command and also ident it)

shannona commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
````diff
@@ -1 +1,85 @@
+    + Execute bitmarkNode-HyperV.bat [Your Public IP]. When run the script without providing [Your Public IP], the script find your IP from Internet
+
+        ``` bitmarkNode-HyperV.bat 117.166.111.123```
+    + Follow script instruction and select "1)Installation"
````
Some comment on the style of how you depict the example command.

shannona commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
````diff
@@ -1 +1,85 @@
+        ``` bitmarkNode-HyperV.bat 117.166.111.123```
+    + Follow script instruction and select "1)Installation"
++ Execute **docker ps** in your command-line/shell to check if bitmark-node is running
+![docker ps result](https://i.imgur.com/l3dF4Hl.jpg)
````
And here, you probably want `docker ps` not **docker ps** for consistency. 

I think the style we've developed elsewhere is `command` or **App Menu**

pieceofr replied:
do you mean ```docker ps``` not  ***docker ps*** in markdown ? 
"I think the style we've developed elsewhere is command or App Menu" ? Do you mean we have a guideline or example?

shannona commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
```diff
@@ -1 +1,85 @@
+When running bitmark node container, user can make sure ports are opened with the following commands.
+
+netcat -v [Your Public IP] 2136
+
```
Does that work on Windows, Linux, and Mac?

pieceofr replied:
Basically, the answer is no for windows but Windows support WSL, I have not tested it.  Only test it on ubuntu/Linux. We assume that users know how to debug their net and we provide them the direction.

pieceofr replied:
find the way to test on windows

shannona commented on learning-bitmark/quick-start/simple-solution-for-node-setup.md:
```diff
@@ -1 +1,85 @@
+netcat -v [Your Public IP] 2136
+
+WebUI is an interface to control bitmark node services. User can only access WebUI through local network. Please notice that user can not access the port from internet due to security reason.
+
```
This feels a little out of place here currently. I think you need to do one of two things:

1) Have a summary at the start of this section that notes that Bitmark uses two ports, 2136 and 9980, but that they serve different purposes and are accessed differently; 

2) or move the 9980 discussion to where it's most appropriate, which is probably "Step 1" above.

shannona reviewed (commented):
Organizationally, this looks fine. I offered a few comments on improving clarity and a tiny bit of standardizing usage.

nodestory committed:
Merge pull request #27 from bitmark-inc/bitmarkd-storage

explain bitmarkd storage

Commit:
- Don't clone the repo/master again

araujobsd committed:
Merge pull request #43 from bitmark-inc/fix-circleci

- Don't clone the repo/master again

nodestory committed:
fix based on igor suggestions

nodestory committed:
Merge pull request #44 from bitmark-inc/improve-storage-style

fix based on igor suggestions

pieceofr committed:
modify based on shannona's review

pieceofr committed:
Merge branch 'quickstartnode' of https://github.com/bitmark-inc/docs into quickstartnode
